### PR TITLE
Introduce diverse conformer generators for both toolkits

### PR DESCRIPTION
**Pull request by mwojcikowski**
A generator which will yield up to `n` conformers for given molecule

### CODE REVIEW

File: oddt/toolkits/ob.py [hunk 1]
```diff
@@ -616,6 +615,71 @@ def __setstate__(self, state):
...
+                 n_conf,  # numConformers
+                 n_conf * 2,  # numChildren
+                 5,  # mutability
+                 5)  # convergence
+        cs.Search()
+        cs.GetConformers(mol_clone.OBMol)
+    elif method == 'confab':
+        ff = pybel._forcefields['uff']
+        ff.Setup(mol_clone.OBMol)
+        ff.DiverseConfGen(0.5, 10000, 5000.0, False)
```
Comment (marta-sd) [inline oddt/toolkits/ob.py]: same here

File: oddt/toolkits/rdk.py [hunk 2]
```diff
@@ -1091,6 +1092,60 @@ def __setstate__(self, state):
...
+    seed : None or int (default=None)
+        Random seed
+
+    Returns
+    -------
+    mols : list of oddt.toolkit.Molecule objects
+        Molecules with diverse conformers
+    """
+    mol_clone = mol.clone
+    if method == 'etkdg':
```
Comment (marta-sd) [inline oddt/toolkits/rdk.py]: And here we can use kwargs

File: oddt/toolkits/rdk.py [hunk 3]
```diff
@@ -1091,6 +1092,60 @@ def __setstate__(self, state):
...
+        params = {'useExpTorsionAnglePrefs': True,
+                  'useBasicKnowledge': False}
+    elif method == 'kdg':
+        params = {'useExpTorsionAnglePrefs': False,
+                  'useBasicKnowledge': True}
+    elif method == 'dg':
+        params = {}
+    else:
+        raise ValueError('Method %s is not implemented' % method)
+    if seed is not None:
```
Comment (marta-sd) [inline oddt/toolkits/rdk.py]: I think it should be `if seed is None`

File: oddt/toolkits/ob.py [hunk 4]
```diff
@@ -616,6 +615,71 @@ def __setstate__(self, state):
...
+    seed : None or int (default=None)
+        Random seed
+
+    Returns
+    -------
+    mols : list of oddt.toolkit.Molecule objects
+        Molecules with diverse conformers
+    """
+    if __version__ < '2.4.0':
+        raise ValueError('Diverse conformer generation is not implemented in '
```
[the remark this answers is not in the document]
Comment (mwojcikowski) [reply oddt/toolkits/ob.py]: Ok

File: oddt/toolkits/ob.py [hunk 5]
```diff
@@ -616,6 +615,71 @@ def __setstate__(self, state):
...
+    mol_clone = mol.clone
+    if seed is not None:
+        rand = ob.OBRandom(True)
+        rand.Seed(seed)
+    if method == 'ga':
+        if not hasattr(ob, 'OBConformerSearch'):
+            raise ValueError('OpenBabel needs to be compiled with eigen to '
+                             'perform conformer search.')
+        cs = ob.OBConformerSearch()
+        cs.Setup(mol_clone.OBMol,
```
[the remark this answers is not in the document]
Comment (mwojcikowski) [reply oddt/toolkits/ob.py]: I've reworked it to use `kwargs`

File: oddt/toolkits/rdk.py (hunk 3 above)
Comment (mwojcikowski) [reply oddt/toolkits/rdk.py]: Oh, yes!!!!111

File: oddt/toolkits/ob.py (hunk 1 above)
Comment (mwojcikowski) [reply oddt/toolkits/ob.py]: Done

File: oddt/toolkits/rdk.py (hunk 2 above)
Comment (mwojcikowski) [reply oddt/toolkits/rdk.py]: I've exposed the RMSD filter to a `rmsd` argument. Which [other parameter](http://www.rdkit.org/Python_Docs/rdkit.Chem.rdDistGeom-module.html#EmbedMultipleConfs) would you like to expose?

Comment (marta-sd) [reply oddt/toolkits/rdk.py]: I think `maxAttempts` and `enforceChirality` might be useful.

**Pull request merged**